libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -987,25 +987,42 @@ void CompilerUtils::convertType(
 			switch (typeOnStack.location())
 			{
 			case DataLocation::Storage:
-				// stack: <source ref>
-				allocateMemory(typeOnStack.memorySize());
-				m_context << Instruction::SWAP1 << Instruction::DUP2;
-				// stack: <memory ptr> <source ref> <memory ptr>
-				for (auto const& member: typeOnStack.members(nullptr))
-				{
-					if (!member.type->canLiveOutsideStorage())
-						continue;
-					pair<u256, unsigned> const& offsets = typeOnStack.storageOffsetsOfMember(member.name);
-					m_context << offsets.first << Instruction::DUP3 << Instruction::ADD;
-					m_context << u256(offsets.second);
-					StorageItem(m_context, *member.type).retrieveValue(SourceLocation(), true);
-					TypePointer targetMemberType = targetType.memberType(member.name);
-					solAssert(!!targetMemberType, "Member not found in target type.");
-					convertType(*member.type, *targetMemberType, true);
-					storeInMemoryDynamic(*targetMemberType, true);
-				}
-				m_context << Instruction::POP << Instruction::POP;
+			{
+				auto conversionImpl = [
+					typeOnStack = dynamic_pointer_cast<StructType const>(_typeOnStack.shared_from_this()),
```

Comment:
Fortunately this will disappear with https://github.com/ethereum/solidity/pull/6417...

Reply:
Why do you need those?

Reply:
Note that the code in the lambda is executed either immediately or not at all, so you can just access the closure via reference.

Reply:
Ah sorry! I confused it with the other one.

Reply:
https://github.com/ethereum/solidity/blob/7c880a26c021fa949f400c9ce5b02a05756bc2a7/libsolidity/codegen/CompilerContext.h#L117